src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -115,7 +121,56 @@ public override async Task<bool> Execute()
                         return false;
                     }
 
-                    installerUpdateList.Add(new InstallerMetadata { InstallerUrl = installerUrl, PackageFile = packageFile });
+                    if (packageFile.IsZipFile())
+                    {
+                        string extractDirectory = Path.Combine(PackageParser.InstallerDownloadPath, Path.GetFileNameWithoutExtension(packageFile));
+
+                        try
+                        {
+                            ZipFile.ExtractToDirectory(packageFile, extractDirectory, true);
+                        }
+                        catch (InvalidDataException invalidDataException)

```

Comment:
>InvalidDataException invalidDataException [](http://example.com/codeflow?start=31&length=41)

Do we want to catch other exceptions and return false too?  For example: PathToolong